lib/galaxy/jobs/__init__.py

```diff
@@ -1298,11 +1298,12 @@ def path_rewriter( path ):
                             dataset.set_peek( is_multi_byte=True )
                         else:
                             dataset.set_peek()
-                    try:
-                        # set the name if provided by the tool
-                        dataset.name = context['name']
-                    except:
-                        pass
+                    for context_key in ['name', 'info', 'dbkey']:
+                        try:
+                            context_value = context[context_key]
+                            setattr(dataset, context_key, context_value)
+                        except Exception:
```

Comment:
It is an `ExpressionContext` not a dictionary - I just looked into the code and it looks like it would throw a `KeyError` and it would respect ` in `. I'l make the change.